python/gsw/main.py

```diff
@@ -0,0 +1,370 @@
+                                return statefield_report
+                        
+                    #if we have not recieved a downlink, return None
+                    return None
+
+    def post_to_es(self):
+        '''
+        Check for the most recent email from iridium. 
+        There are two conditions under which we would create an Iridium report: 
+        1) When we recieve a statefield report
+        2) when we recieve an uplink confirmation.
+        - If there is a statefield report, index the statefield report and 
+        create and index an iridium report.
+        - If there is not a statefield report but we recently recieved an uplink
+        confirmation, then create and index an iridium report.
+        '''
+
+        while self.run_email_thread==True:
+            #get the most recent statefield report
+            sf_report=readIr.check_for_email()
+
+            if sf_report is not None:
+                # Print the statefield report recieved
+                print("Got report: "+str(sf_report)+"\n")
+
+                # Index statefield report in elasticsearch
+                statefield_res = self.es.index(index='statefield_report_'+str(self.imei), doc_type='report', body=sf_report)
+                # Print whether or not indexing was successful
+                print("Statefield Report Status: "+statefield_res['result'])
+
+                # Create an iridium report 
+                ir_report=json.dumps({
+                    "momsn":self.momsn,
+                    "mtmsn":self.mtmsn, 
+                    "confirmation-mtmsn": self.confirmation_mtmsn,
+                    "send-uplinks": self.send_uplinks,
+                    "time": str(datetime.now().isoformat())
+                })
+
+                # Index iridium report in elasticsearch
+                iridium_res = self.es.index(index='iridium_report_'+str(self.imei), doc_type='report', body=ir_report)
+                # Print whether or not indexing was successful
+                print("Iridium Report Status: "+iridium_res['result']+"\n\n")
+
+            elif self.recieved_uplink_confirmation:
+                # Create an iridium report and add that to the iridium_report index in elasticsearch. 
+                ir_report=json.dumps({
+                    "momsn":self.momsn,
+                    "mtmsn":self.mtmsn, 
+                    "confirmation-mtmsn": self.confirmation_mtmsn,
+                    "send-uplinks": self.send_uplinks,
+                    "time": str(datetime.now().isoformat())
+                })
+
+                # Index iridium report in elasticsearch
+                iridium_res = self.es.index(index='iridium_report_'+str(self.imei), doc_type='report', body=ir_report)
+                # Print whether or not indexing was successful
+                print("Iridium Report Status: "+iridium_res['result']+"\n\n")
+                # Record that we have not recently recieved any uplinks as we just indexed the most recent one
```

Comment:
There is duplicated Iridium report code here. I think it would be better to restructure like this:

- Add Iridium report, always
- Add state field report if it exists
- If uplink confirmation is received, set the flag to true

Reply:
Do you mean we should be adding an Iridium report every 10 seconds? I still think it makes more sense to make iridium reports whenever we receive either a downlink or an uplink confirmation, since those are the only situations in which the momsn and mtmsn numbers would be changed.

Reply:
Oops!  I misread the code, and you're right, I think that does make more sense. I still think we should break out the redundant code into a function, though.